src/installer/tests/HostActivation.Tests/NativeHostApis.cs

```diff
@@ -243,6 +269,213 @@ public void Hostfxr_corehost_set_error_writer_test()
+                .And.HaveStdOutContaining($"hostfxr_get_dotnet_environment_info sdk versions:[{expectedSdkVersions}]")
+                .And.HaveStdOutContaining($"hostfxr_get_dotnet_environment_info sdk paths:[{expectedSdkPaths}]")
+                .And.HaveStdOutContaining($"hostfxr_get_dotnet_environment_info framework names:[{expectedFrameworkNames}]")
+                .And.HaveStdOutContaining($"hostfxr_get_dotnet_environment_info framework versions:[{expectedFrameworkVersions}]")
+                .And.HaveStdOutContaining($"hostfxr_get_dotnet_environment_info framework paths:[{expectedFrameworkPaths}]");
+            }
+        }
+
+        [Fact]
+        public void Hostfxr_get_dotnet_environment_info_with_multilevel_lookup_only()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // Only Windows supports multi-level lookup.
+                return;
+            }
+
+            var f = new SdkResolutionFixture(sharedTestState);
+            string expectedSdkVersions = string.Join(';', new[]
+            {
+                "1.2.3",
+                "2.3.4-preview",
+                "3.0.0",
+                "4.5.6",
+                "5.6.7",
+                "15.1.4-preview"
+            });
+
+            string expectedSdkPaths = string.Join(';', new[]
+            {
+                Path.Combine(f.ProgramFilesGlobalSdkDir, "1.2.3"),
+                Path.Combine(f.ProgramFilesGlobalSdkDir, "2.3.4-preview"),
+                Path.Combine(f.SelfRegisteredGlobalSdkDir, "3.0.0"),
+                Path.Combine(f.ProgramFilesGlobalSdkDir, "4.5.6"),
+                Path.Combine(f.SelfRegisteredGlobalSdkDir, "5.6.7"),
+                Path.Combine(f.SelfRegisteredGlobalSdkDir, "15.1.4-preview"),
+            });
+
+            string expectedFrameworkNames = string.Join(';', new[]
+            {
+                "HostFxr.Test.A",
+                "HostFxr.Test.A",
+                "HostFxr.Test.B",
+            });
+
+            string expectedFrameworkVersions = string.Join(';', new[]
+            {
+                "1.2.3",
+                "3.0.0",
+                "5.6.7-A",
+            });
+
+            string expectedFrameworkPaths = string.Join(';', new[]
+            {
+                Path.Combine(f.ProgramFilesGlobalFrameworksDir, "HostFxr.Test.A"),
+                Path.Combine(f.ProgramFilesGlobalFrameworksDir, "HostFxr.Test.A"),
+                Path.Combine(f.ProgramFilesGlobalFrameworksDir, "HostFxr.Test.B"),
+            });
+
+            using (TestOnlyProductBehavior.Enable(f.Dotnet.GreatestVersionHostFxrFilePath))
+            {
+                // We pass f.WorkingDir so that we don't resolve dotnet_dir to the global installation
+                // in the native side.
+                f.Dotnet.Exec(f.AppDll, new[] { "hostfxr_get_dotnet_environment_info", f.WorkingDir })
```

Comment:
We should also have a test where the `dotnet_root` is `null` - to validate that code path.